derp/limiter.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package derp
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func readTxBytes(interfaceName string) (uint64, error) {
+	v, err := ioutil.ReadFile("/sys/class/net/" + interfaceName + "/statistics/tx_bytes")
```

Comment:
Doing it this way means that if I'm scp-ing something off a derp server, all clients on the server will get throttled. I don't think that's what we want? I'd strongly prefer that we do the accounting of sent bytes within derper, rather than rely on undifferentiated OS stats.

Plus this means derper can't run on !linux any more, which previously was possible.